### PR TITLE
Create UI Flow

### PR DESCRIPTION
**Pull request by returningsam**

### CODE REVIEW

File: src/main/java/rpgcharacters/RpgCharacters.java [hunk 1]
```diff
@@ -54,7 +53,8 @@ public static void main(String[] args) {
 
         // Output usage if no args or help option input
         if (args.length == 0 || rpg.help) {
-            jc.usage();
```
Comment (mwitte98) [inline src/main/java/rpgcharacters/RpgCharacters.java]: Move `if (rpg.help) jc.usage()` to a new if statement and update comment

**Comment by mwitte98** [comment]
Change userFlow folder name to userflow

**Comment by mwitte98** [comment]
When any character other than a number is input for a menu, `java.util.InputMismatchException` is thrown and crashes the program

**Comment by mwitte98** [comment]
When a party's quests are listed, the next input says `Please enter the number of the desired party here:`, but party should be quest

File: src/main/java/rpgcharacters/userFlow/PartyMenu.java [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+    */
+    public void enter () {
+        printMenuTitle();
+        int input;
+        String party;
+        int exit = 6;
+        do {
+
+            printOptions();
+            input = sc.nextInt();
```
Comment (mwitte98) [inline src/main/java/rpgcharacters/userFlow/PartyMenu.java]: We need a `try` around this one too

File: src/main/java/rpgcharacters/userFlow/InitMenu.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+        );
+        System.out.print("Please enter the number of the desired option here: ");
+    }
+
+    /**
+    * Defines the loop for this menu
+    */
+    public void enter () {
+        printMenuTitle();
+        int input;
```
Comment (mwitte98) [inline src/main/java/rpgcharacters/userFlow/InitMenu.java]: `input` needs to be set to something in all 5 files with a `try`, maybe set to 0?

`mvn package` succeeds but I get this error when trying to run the jar:
```
Exception in thread "main" java.lang.Error: Unresolved compilation problem: 
	The local variable input may not have been initialized

	at rpgcharacters.userflow.InitMenu.enter(InitMenu.java:65)
	at rpgcharacters.RpgCharacters.main(RpgCharacters.java:57)
```

File: src/main/java/rpgcharacters/RpgCharacters.java (hunk 1 above)
Comment (mwitte98) [inline src/main/java/rpgcharacters/RpgCharacters.java]: @returningsam the help comment is here 😉

File: src/main/java/rpgcharacters/userFlow/PartyAddCharMenu.java [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+        //       to the userNames ArrayList.
+
+        charNames.add("Bilbo Baggins");
+        charNames.add("John Cena");
+        charNames.add("Rick C137");
+        userNames.add("idkmanthings");
+        userNames.add("username");
+        userNames.add("souperPerson8");
+
+        if (charNames.size() != userNames.size()) {
```
Comment (ekliot) [inline src/main/java/rpgcharacters/userFlow/PartyAddCharMenu.java]: would it make more sense to make a hash table of `<character, username>`?

File: src/main/java/rpgcharacters/userFlow/PartyRemCharMenu.java [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+        //       to the userNames ArrayList.
+
+        charNames.add("Bilbo Baggins");
+        charNames.add("John Cena");
+        charNames.add("Rick C137");
+        userNames.add("idkmanthings");
+        userNames.add("username");
+        userNames.add("souperPerson8");
+
+        if (charNames.size() != userNames.size()) {
```
Comment (ekliot) [inline src/main/java/rpgcharacters/userFlow/PartyRemCharMenu.java]: is a hash map better here?

File: src/main/java/rpgcharacters/userFlow/PartyMenu.java [hunk 6]
```diff
@@ -50,14 +50,63 @@ private String printParties () {
...
+
+        System.out.println(
+            "--------------------------------------------------\n" + // 50 chars
+            partyName + "\n" +
+            "--------------------------------------------------\n" + // 50 chars
+            "Game master: " + gameMaster + "\n" +
+            "Characters:"
+        );
+
+        if (charNames.size() != userNames.size()) {
```
Comment (ekliot) [inline src/main/java/rpgcharacters/userFlow/PartyMenu.java]: is a hash map better here?

**Pull request merged**